esrally/client.py

```diff
@@ -26,6 +26,87 @@
 from esrally.utils import console, convert
 
 
+class RequestContextManager:
+    """
+    Ensures that request context span the defined scope and allow nesting of request contexts with proper propagation.
+    This means that we can span a top-level request context, open sub-request contexts that can be used to measure
+    individual timings and still measure the proper total time on the top-level request context.
+    """
+    def __init__(self, client):
+        self.client = client
```

Comment:
Initially unclear to me that client here is a ref to the RequestContextHolder. A small comment would of made this easier to follow